src/target/adiv5.c

```diff
@@ -1072,6 +1072,41 @@ void firmware_mem_write_sized(adiv5_access_port_s *ap, uint32_t dest, const void
 	adiv5_dp_read(ap->dp, ADIV5_DP_RDBUFF);
 }
 
+void mm32l0_firmware_mem_write_sized(adiv5_access_port_s *ap, uint32_t dest, const void *src, size_t len, align_e align)
+{
+	uint32_t odest = dest;
+
+	len >>= align;
+	ap_mem_access_setup(ap, dest, align);
+	while (len--) {
+		uint32_t tmp = 0;
+		/* Pack data into correct data lane */
+		switch (align) {
```

Comment:
Please use the switch-case style of adiv5_pack_data() here. It results in better code and avoids an undefined behaviour (UB) cast + unaligned access to src in the ALIGN_DWORD and ALIGN_WORD cases. (we are aware of the contents of firmware_mem_write_sized(), it needs replacing after renaming.. it's duplicating adiv5_pack_data() for no good reason)

It looks like the only change this needs from that code is to replace ALIGN_HALFWORD with:
```c
uint16_t value;
memcpy(&value, src, sizeof(value));
tmp = value;
```
This will already automatically widen correctly because the types all agree on being unsigned, etc.